hvac/v1/__init__.py

```diff
@@ -978,6 +986,156 @@ def auth_approle(self, role_id, secret_id=None, mount_point='approle', use_token
 
         return self.auth('/v1/auth/{0}/login'.format(mount_point), json=params, use_token=use_token)
 
+    def auth_iam(self, iam_role):
+        """
+        Authenticate to Vault per https://www.vaultproject.io/docs/auth/aws.htm.
+
+        :param iam_role: IAM role
+        :return:
+        """
+        parsed = urlparse(self._url)
+        self.token = self._authenticate_to_vault(parsed.hostname,
+                                                 parsed.port,
+                                                 iam_role,
+                                                 self._kwargs['verify'])
+        return self.token
+
+    def _authenticate_to_vault(self, vault_host, vault_port, role, verify):
+        payload = self._generate_vault_request(role, vault_host)
+
+        headers = {
+            'Content-type': 'application/json',
+            'Accept': 'text/plain'  # ,
+        }
+
+        response = requests.post('https://{}:{}/v1/auth/aws/login'.format(vault_host, vault_port),
+                                 data=json.dumps(payload),
+                                 headers=headers,
+                                 verify=verify)
+        if response.status_code != 200:
+            raise Exception(
+                "Failed to login to Vault due to error {} with body {}".format(response.status_code, response.text))
+
+        body = response.json()
+        return body['auth']['client_token']
+
+    def _generate_vault_request(self, role, vault_host):
+        """
+        Generate a signed sts:GetCallerIdentity request to validate identify of the client.
+        The Vault server reconstructs the query and forwards it to STS service to authenticate
+        the client. See https://www.vaultproject.io/docs/auth/aws.html for more.
+
+        :param role: Role of this lambda
+        :return: Request body
+        """
+
+        client = boto3.client('sts')
```

Comment:
I personally would rather have any code that needs to lean on boto3 be handled outside of this client. If someone has cause to use hvac but doesn't need any of the AWS features, its not fair to expect them to install a boto dependency. The boto library is only being used for a couple of class in this PR, so leaving those calls to be handled by the caller instead of this library is not asking too much IMHO.